syntax = "proto3";

enum AppliedAction {
	ENCODE=0;
	HASH=1;
	SIGN=2;
	ENCRYPT_ASYMMETRIC=3;
	ENCRYPT_SYMMETRIC=4;
	CERTIFICATE=5;
}

message Payload {
	AppliedAction action = 1;
	string payload_str = 2;
	string original_message = 3;
	map<string,string> metadata = 4;	// Additional data
}

